src/widget/wtrackmenu.cpp

```diff
@@ -0,0 +1,1548 @@
+            return;
+        }
+    }
+
+    // TODO(XXX): Care whether the append succeeded.
+    m_pTrackCollectionManager->unhideTracks(trackIds);
+    playlistDao.appendTracksToPlaylist(trackIds, iPlaylistId);
+}
+
+
+void WTrackMenu::slotPopulateCrateMenu() {
+    // The user may open the Crate submenu, move their cursor away, then
+    // return to the Crate submenu before exiting the track context menu.
+    // Avoid querying the database multiple times in that case.
+    if (m_bCrateMenuLoaded) {
+        return;
+    }
+    m_pCrateMenu->clear();
+    const TrackIdList trackIds = m_pTrackIdList;
+
+    CrateSummarySelectResult allCrates(m_pTrackCollectionManager->internalCollection()->crates().selectCratesWithTrackCount(trackIds));
+
+    CrateSummary crate;
+    while (allCrates.populateNext(&crate)) {
+        auto pAction = make_parented<QWidgetAction>(m_pCrateMenu);
+        auto pCheckBox = make_parented<QCheckBox>(m_pCrateMenu);
+
+        pCheckBox->setText(crate.getName());
+        pCheckBox->setProperty("crateId",
+                                QVariant::fromValue(crate.getId()));
+        pCheckBox->setEnabled(!crate.isLocked());
+        // Strangely, the normal styling of QActions does not automatically
+        // apply to QWidgetActions. The :selected pseudo-state unfortunately
+        // does not work with QWidgetAction. :hover works for selecting items
+        // with the mouse, but not with the keyboard. :focus works for the
+        // keyboard but with the mouse, the last clicked item keeps the style
+        // after the mouse cursor is moved to hover over another item.
+
+        // ronso0 Disabling this stylesheet allows to override the OS style
+        // of the :hover and :focus state.
+//        pCheckBox->setStyleSheet(
+//            QString("QCheckBox {color: %1;}").arg(
+//                    pCheckBox->palette().text().color().name()) + "\n" +
+//            QString("QCheckBox:hover {background-color: %1;}").arg(
+//                    pCheckBox->palette().highlight().color().name()));
+        pAction->setEnabled(!crate.isLocked());
+        pAction->setDefaultWidget(pCheckBox.get());
+
+        if (crate.getTrackCount() == 0) {
+            pCheckBox->setChecked(false);
+        } else if (crate.getTrackCount() == (uint)trackIds.length()) {
+            pCheckBox->setChecked(true);
+        } else {
+            pCheckBox->setTristate(true);
+            pCheckBox->setCheckState(Qt::PartiallyChecked);
+        }
+
+        m_pCrateMenu->addAction(pAction.get());
+        connect(pAction.get(), &QAction::triggered,
+                this, [this, pCheckBox{pCheckBox.get()}] { updateSelectionCrates(pCheckBox); });
+        connect(pCheckBox.get(), &QCheckBox::stateChanged,
+                this, [this, pCheckBox{pCheckBox.get()}] { updateSelectionCrates(pCheckBox); });
+
+    }
+    m_pCrateMenu->addSeparator();
+    QAction* newCrateAction = new QAction(tr("Create New Crate"), m_pCrateMenu);
```

Comment:
```suggestion
    QAction* newCrateAction = new QAction(tr("Add to New Crate"), m_pCrateMenu);
```
The current description does not make it clear that the selected tracks will be added to the new crate.

Reply:
True. I will change this.